core/app/[locale]/(default)/cart/page-data.ts

```diff
@@ -194,3 +260,25 @@ export const getCart = async (variables: Variables) => {
 
   return data;
 };
+
+export const getShippingCountries = async (geography: FragmentOf<typeof GeographyFragment>) => {
+  const hasAccessToken = Boolean(process.env.BIGCOMMERCE_ACCESS_TOKEN);
+  const shippingZones = hasAccessToken ? await getShippingZones() : [];
+  const countries = geography.countries ?? [];
+
+  const uniqueCountryZones = shippingZones.reduce<Set<string>>((zones, item) => {
+    item.locations.forEach(({ country_iso2 }) => {
+      zones.add(country_iso2);
+    });
+
+    return zones;
+  }, new Set());
+
+  const uniqueCountryZonesArray = Array.from(uniqueCountryZones);
+
+  return countries.filter((countryDetails) => {
+    const isCountryInTheList = uniqueCountryZonesArray.includes(countryDetails.code);
+
+    return isCountryInTheList || !hasAccessToken;
+  });
```

Comment:
You'll need to test this, but I think you can make this more simple by using the set directly instead of reducing on a set
```suggestion
  const hasAccessToken = Boolean(process.env.BIGCOMMERCE_ACCESS_TOKEN);
  const shippingZones = hasAccessToken ? await getShippingZones() : [];
  const countries = geography.countries ?? [];

  const uniqueCountryZones = new Set(shippingZones.map(zone => zone.locations.map(location => location.country_iso2)).flat());

  return countries.filter((countryDetails) => {
    const isCountryInTheList = uniqueCountryZones.has(countryDetails.code);

    return isCountryInTheList || !hasAccessToken;
  });
```

Reply:
Tested this and it works. Updated! Thanks for the suggestion.